ui/pages/confirmations/components/confirm/info/contexts/advanced-details-context.tsx

```diff
@@ -0,0 +1,45 @@
+import React, { ReactElement, createContext, useContext, useMemo } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { getConfirmationAdvancedDetailsOpen } from '../../../../../../selectors';
+import { setConfirmationAdvancedDetailsOpen } from '../../../../../../store/actions';
+
+type AdvancedDetailsHandlerContextType = {
+  showAdvancedDetails: boolean;
+  setShowAdvancedDetails: (value: boolean) => void;
+};
+
+export const AdvancedDetailsHandlerContext = createContext<
+  AdvancedDetailsHandlerContextType | undefined
+>(undefined);
+
+export const AdvancedDetailsProvider: React.FC<{
+  children: ReactElement;
+}> = ({ children }) => {
+  const dispatch = useDispatch();
+
+  const showAdvancedDetails = useSelector(getConfirmationAdvancedDetailsOpen);
+  const setShowAdvancedDetails = (value: boolean): void => {
+    dispatch(setConfirmationAdvancedDetailsOpen(value));
+  };
+
+  const advancedDetailsObject = useMemo(
+    () => ({ showAdvancedDetails, setShowAdvancedDetails }),
+    [showAdvancedDetails, setShowAdvancedDetails],
+  );
```

Comment:
useMemo here does not looks very useful.

Reply:
The `useMemo` is primarily used for maintaining referential equality and potentially preventing unnecessary re-renders in child components.

Reply:
I think the container for the properties typically shouldn't matter, as any dependent component or hook would be using the values within it the object as the dependencies, rather than the container itself.